Overview and Resource Safety section of documentation (#136)
monix/monix-bio: docs/overview.md

If you are already familiar with `Task` - learning `BIO` is straightforward because the only difference is in
error handling - the rest of API is the same. In many cases, migration might be as simple as changing imports from `monix.eval.Task` to `monix.bio.Task`.

To me, the big difference between Monix and other effect libraries is approach to impure code.
Both `IO` and `ZIO` will push you to write 100% purely functional codebase, except for isolated cases where low-level imperative code is needed for performance.
Monix favors purely functional programming too (the only impure method is `memoize`), but we also recognize and fully support users who would rather go for hybrid approach, or are allergic to pure FP.
I feel like it plays to Scala's unique strengths. 
Few examples of Monix providing extra support for `Future` users:
- `monix-execution` module provides many utilities to use with `Future` even if you're not interested in `Task` at all.
- Monix uses `Scheduler` which is also `ExecutionContext` and can be used with `Future` directly. 
- `Local` works with both `Future` and Monix `Task/BIO`. 

> ## Other effects

Reviewer: This section might be in a better place at the end of the documentation? Just to make it shorter and let the user to go faster to hands on.
They could just skip it if they want to, but if I were the reader, I would expect to see some motivation at the start, and then at the end the comparison with other effects. 
Of course this is just personal opinion, if you prefer it here don't bother changing it ;)

Author: Do you mean to put it on a different page?

Reviewer: Yup, I meant that, at the end of the doc indeed.

Author: I like the idea, thanks for the suggestion

title: Overview
---

`BIO[E, A]` represents a specification for a possibly lazy or asynchronous computation. 
When executed, it will produce a successful value `A`, an error `E`, never terminate or complete with a terminal (untyped) error.

`BIO` handles concurrency, cancellation, resource safety, context propagation, error handling, and can suspend effects.

> All of it makes it a good tool to write a simple, high-level code to solve problems related to any of these features in a safe and performant manner.

Reviewer 2: ```suggestion
All of this makes it simple to write good, high-level code that solve problems related to any of these features in a safe and performant manner.
```

DISCLAIMER: The following sections are very subjective and biased opinions of the author.

There are already many effect types in Scala, i.e. [cats.effect.IO](https://github.com/typelevel/cats-effect), [Monix Task](https://github.com/monix/monix), and [ZIO](https://github.com/zio/zio).
It begs a question - why would anyone want another one?

Well, it seems like built-in typed errors are in great demand, and the only other effect which has built-in typed errors is `ZIO`. 
Not everyone wants to use `ZIO` and I feel like there are enough differences in `Monix` to make it a valuable alternative.
The target audience of BIO are users of `IO`, `Task`, and `Future` who tend to use `EitherT` a lot and would like to have a smoother experience, with better type inference, no syntax imports, and without constant wrapping and unwrapping.

`Monix BIO` builds upon `Monix Task` and enhances it with typed error capabilities.
If you are already familiar with `Task` - learning `BIO` is straightforward because the only difference is in
error handling - the rest of API is the same. In many cases, migration might be as simple as changing imports from `monix.eval.Task` to `monix.bio.Task`.

> To me, the big difference between Monix and other effect libraries is approach to impure code.

Reviewer 2: ```suggestion
To me, the big difference between Monix and other effect libraries is it's approach to impure code.
```

> Both `IO` and `ZIO` will push you to write 100% purely functional codebase, except for isolated cases where low-level imperative code is needed for performance.

Reviewer 2: ```suggestion
Both `IO` and `ZIO` will push you to write a 100% purely functional codebase, except for isolated cases where low-level imperative code is needed for performance.
```

> Monix favors purely functional programming too (the only impure method is `memoize`), but we also recognize and fully support users who would rather go for hybrid approach, or are allergic to pure FP.

Reviewer 2: ```suggestion
Monix favors purely functional programming too (the only impure method is `memoize`), but we also recognize and fully support users who would rather go for a hybrid approach, or are allergic to pure FP.
```

I feel like it plays to Scala's unique strengths. 

> Few examples of Monix providing extra support for `Future` users:

Reviewer 2: ```suggestion
Here are a few examples of Monix providing extra support for users of `Future`:
```

> - `monix-execution` module provides many utilities to use with `Future` even if you're not interested in `Task` at all.

Reviewer 2: ```suggestion
- The `monix-execution` module provides many utilities to use with `Future` even if you're not interested in `Task` at all.
```

> - Monix uses `Scheduler` which is also `ExecutionContext` and can be used with `Future` directly. 

Reviewer 2: ```suggestion
- Monix uses a `Scheduler` which is also an `ExecutionContext` and can be used with `Future` directly. 
```
